expyriment/design/_structure.py

```diff
@@ -1792,15 +1792,17 @@ def find_trial(self, id):
     def copy(self):
         """Return a copy of the block."""
 
-        owntrials = []
-        triallist = []
-        for trial in self._trials:
-            owntrials.append(trial)
-            triallist.append(trial.copy())
+        # reset self._trials to avoid deep-copying it
+        trials = self._trials
         self._trials = []
+
+        # deep-copy everything except self._trials
         rtn = deepcopy(self)
-        self._trials = owntrials
-        rtn._trials = triallist
+        self._trials = trials
+
+        # manually half-deep-copy self._trials
+        rtn._trials = [trial.copy() for trial in trials]
```

Comment:
Note that `Trial.copy()` does **not** perform a deep copy of the stimuli, see below.

Reply:
I understand now this is on purpose: you want a shallow copy of the list of stimuli, but not a copy of the stimuli themselves.